docs/language/ql-handbook/types.rst

```diff
@@ -479,6 +480,53 @@ program, so it's helpful to extend a new type (namely ``TTaintType``)::
     class Tainted extends TaintType, TTaintedValue {
     }
     
+.. _type-unions:
+
+Type unions
+***********
+
+.. note::
+    The syntax for type unions is considered experimental and is subject to change.
```

Comment:
I wouldn't call this syntax experimental. Are we really not committed to it?

Reply:
I added this note (based on the note from algebraic datatypes) because of my understanding that this was an internal-only feature currently. I suppose that's not actually what the wording says though, I will remove it.